client/app/users/login/login.component.ts

```diff
@@ -0,0 +1,102 @@
+import { Component, OnInit } from '@angular/core';
+import { ActivatedRoute, Router } from '@angular/router';
+import { filter, map, switchMap } from 'rxjs/operators';
+import { UsersService } from '../users.service';
+
+@Component({
+    selector: 'app-login',
+    templateUrl: './login.component.html',
+    styleUrls: ['./login.component.css']
+})
+export class LoginComponent implements OnInit {
+    /**
+     * Current entered pin.
+     */
+    pin = '';
+
+    /**
+     * Determine if there has been an error with the current pin.
+     */
+    hasError = false;
+
+    /**
+     * Whether to clear the input on the next char entered.
+     */
+    clearOnNext = false;
+
+    /**
+     * The url to redirect the user after a successful pin entry.
+     */
+    url = '/';
+
+    get splitPin() {
+        return new Array(this.pin.length);
+    }
+
+    constructor(
```

Comment:
Comment for this function. 
The bool values above could be set as `hasError`: Boolean = false or just Boolean; and then set it to false on ngOnInit(). Not required tho